src/h3lib/include/polygon.h

```diff
@@ -24,13 +24,61 @@
 #include "bbox.h"
 #include "geoCoord.h"
 #include "h3api.h"
+#include "linkedGeo.h"
+
+/** @struct IterableGeoLoop
+ *  @brief  Struct supporting polymorphic loop iteration
+ */
+typedef struct {
+    union {
```

Comment:
Can I just name this within the struct?

Reply:
Well the easiest thing is not to name the type, but name the field. So `union {...} value;` is valid. Just use `x.value.geofence` instead of `x.geofence` when you need to access the inner fields.

Reply:
Is there a syntactical way to do this otherwise? And do we care? None of our builds are telling me that this isn't valid.

Reply:
If you want to rely on this format you might want to update toolchain.cmake to use `set(CMAKE_C_EXTENSIONS ON)` because iirc this is a GNU extension.